### ISSUE TITLE
[0.9.3] write timeouts

### ISSUE THREAD

**Issue by jhorwit2** [influxdb/influxdb#3899]
I have been running 0.9.3 with the same point rate (2 thousand every second or two) since release day. Today I started noticing write timeouts with really long wal compaction times, which i'm unsure if that's the issue.

`[wal] 2015/08/30 12:25:38 compaction of partition 1 took 1m32.283077624s`

I tried several restarts, but timeouts start again immediately. My wal directory is only 2mb in size.

**Comment by dswarbrick** [comment]
@jhorwit2 Take a look at #3885 too.

**Comment by jhorwit2** [comment]
Thanks @dswarbrick I'll keep an eye on that issue. My database didn't start go south until ~22GB which is about double when yours did.

**Comment by jhorwit2** [comment]
I'm also not running into OOM issues. My other smaller databases write just fine, but my larger database fails 100% of the time.

**Comment by dswarbrick** [comment]
@jhorwit2 After several crashes and restarts, without nuking the DB in between, my DB is now at 21 GB. The last time I restarted InfluxDB, I nuked the WAL directory, but it has now filled the 2 GB tmpfs that it resides on, and I'm once again getting write timeouts.

My gut feeling is that BoltDB is not only writing to the most recent shard, but repeatedly touching the entire 21 GB DB, and this only gets worse as the DB grows. For us at least, InfluxDB is unusable in its current state.
